app/ts/background/windows/personalSign.ts

```diff
@@ -30,9 +30,17 @@ export async function resolvePersonalSign(websiteTabConnections: WebsiteTabConne
 		const resolved = await resolve(confirmation, data.simulationMode, data.params)
 		postMessageIfStillConnected(websiteTabConnections, data.socket, { ...data.params, ...resolved, requestId: confirmation.data.requestId })
 	}
+	if (openedDialog) await closePopupOrTab(openedDialog)
 	openedDialog = undefined
 }
 
+export async function updatePendingPersonalSignViewWithPendingRequests(ethereumClientService: EthereumClientService) {
+	const request = await getPendingPersonalSignPromise()
+	if (request != undefined) {
```

Comment:
looks like not all conditions are returned from this function?

Reply:
it returns void so it doesn't need to return in all conditions